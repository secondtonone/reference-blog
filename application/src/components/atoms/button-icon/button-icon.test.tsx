import renderer from 'react-test-renderer';
import ButtonIcon from '.';
import SvgIcon from '~/components/containers/svg-icon';

jest.mock('~/components/containers/svg-icon', () => jest.fn().mockReturnValue('svg-icon'));

describe('Match snapshots ButtonIcon', () => {
  it('with up icon', () => {
    const tree = renderer.create(<ButtonIcon code="up" />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(SvgIcon).toHaveBeenCalledWith({ code: 'up' }, {});
  });

  it('with up icon and view="filled"', () => {
    const tree = renderer.create(<ButtonIcon code="up" view="filled" />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(SvgIcon).toHaveBeenCalledWith({ code: 'up' }, {});
  });
});
