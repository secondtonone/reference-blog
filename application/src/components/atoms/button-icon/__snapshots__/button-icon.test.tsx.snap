// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshots ButtonIcon with up icon 1`] = `
<span
  className="base-button__BaseButton-sc-1hqbwl-0 secondary-button__SecondaryButton-sc-1lq79t2-0 dGmWiH PcQft button-icon__ButtonIconStyled-sc-i4ys4z-0 czAHmw"
  data-test="button-icon-up"
  size="m"
>
  <button
    className="___SButton_rkzvx_gg_ _size_m_rkzvx_gg_ _theme_secondary-muted_rkzvx_gg_"
    data-ui-name="Button"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    size="m"
    style={Object {}}
    tabIndex={0}
    type="button"
  >
    <span
      className="___SInner_rkzvx_gg_"
      data-ui-name="Box"
      style={Object {}}
    >
      <span
        className="_size_m_rkzvx_gg_ ___SText_rkzvx_gg_"
        data-ui-name="Button.Text"
        size="m"
        style={Object {}}
      >
        svg-icon
      </span>
    </span>
  </button>
</span>
`;

exports[`Match snapshots ButtonIcon with up icon and view="filled" 1`] = `
<span
  className="base-button__BaseButton-sc-1hqbwl-0 secondary-button__SecondaryButton-sc-1lq79t2-0 dGmWiH PcQft button-icon__ButtonIconStyled-sc-i4ys4z-0 dNfjhV"
  data-test="button-icon-up"
  size="m"
>
  <button
    className="___SButton_rkzvx_gg_ _size_m_rkzvx_gg_ _theme_secondary-muted_rkzvx_gg_"
    data-ui-name="Button"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    size="m"
    style={Object {}}
    tabIndex={0}
    type="button"
  >
    <span
      className="___SInner_rkzvx_gg_"
      data-ui-name="Box"
      style={Object {}}
    >
      <span
        className="_size_m_rkzvx_gg_ ___SText_rkzvx_gg_"
        data-ui-name="Button.Text"
        size="m"
        style={Object {}}
      >
        svg-icon
      </span>
    </span>
  </button>
</span>
`;
